specs/saltpack_armor.md

```diff
@@ -255,31 +255,46 @@ ipsum](https://twitter.com/oconnor663/status/680171387353448448).
 
 ## Framing the BaseX Payload
 
-Before getting to the BaseX payload, the decoder parses the header and footer:
+The BaseX payload is surrounded ('framed') by a header and footer. Rules for parsing
+them appear here.
 
-1. Strip all whitespace and `>` characters from the message. (`>` is for
-   compatibility with email clients that use it for quoting.)
-2. Collect input up to the first period. This is the header.
-3. Assert that the header (with its whitespace stripped in step 1) matches
-
-   ```
-   BEGIN([a-zA-Z0-9]+)?SALTPACK(ENCRYPTEDMESSAGE)|(SIGNEDMESSAGE)|(DETACHEDSIGNATURE)
-   ```
-
-   The optional word is for an application name (like `KEYBASE`). The last two
-   words give the mode of the message.
-4. Collect input up to the second period. This is the payload. If the
-   implementation is streaming, it may decode the payload before the following
-   steps.
+0. For this section, consider the "stripping set" to be the union of all whitespace and
+   the `>` character (for compatibility with mail clients).
+1. Collect input up to the first period. This is the header.
+2. Within the header, convert all blocks of characters in the stripping set to single spaces.
+   That is to say, any contiguous substring of the header where all characters are in the
+   stripping set should be converted to a single space. Then, strip any leading or trailing
+   whitespace. For example, the string
+```
+\n> BEGIN KEYBASE      SALTPACK\n   ENCRYPTED\t\t    MESSAGE
+```
+   should become
+```
+BEGIN KEYBASE SALTPACK ENCRYPTED MESSAGE
+```.
+2. Convert all whitespace to a single space for easy parsing. For example, in Go we match with
+   the regex `[>\\s]+`, which catches all characters in the union of whitespace
+   and `>` (`>` is for compatibility with mail clients that use it for quoting).
```

Comment:
I think this section has accidentally gotten left in. Let's go over the rest in person.